pkg/etcd/etcd.go

```diff
@@ -892,3 +895,61 @@ func backupDirWithRetention(dir string, maxBackupRetention int) (string, error)
 	}
 	return backupDir, nil
 }
+
+// GetAPIServerURLFromETCD will try to fetch the version.Program/apiaddresses key from etcd
+// when it succeed it will parse the first address in the list and return back an IP and Port
+func GetAPIServerURLFromETCD(ctx context.Context, cfg *config.Control) (string, int, error) {
```

Comment:
Just make this return (string, error). You split it by a semicolon here so that you can return the IP and Port separately, and then just end up calling `string.Sprintf()` to put the IP and Port back together in a string later on. Might as well just keep it as a string all the way through and avoid the unnecessary split/format.

Reply:
fixing, thanks!

Reply:
fixed